modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache

```diff
@@ -1,12 +1,24 @@
 {{>partial_header}}
 
+use serde::Deserialize;
 {{#withAWSV4Signature}}
 use std::time::SystemTime;
 use aws_sigv4::http_request::{sign, SigningSettings, SigningParams, SignableRequest};
 use http;
 use secrecy::{SecretString, ExposeSecret};
 {{/withAWSV4Signature}}
 
+#[derive(Debug, Clone, Deserialize)]
+pub struct DeserializedConfiguration {
+    pub base_path: String,
+    pub user_agent: Option<String>,
+    pub basic_auth: Option<BasicAuth>,
+    pub oauth_access_token: Option<String>,
+    pub bearer_access_token: Option<String>,
+    pub api_key: Option<ApiKey>,
+    // TODO: take an oauth2 token source, similar to the go one
+}
```

Comment:
This allows users to read configuration from a file, which I imagine is a common use case. 

Some of the structs used in config don't implement Deserialize - thinking of `Secret` for example. The properties of those types cannot be decorated with `#[serde(skip)]`  either as they don't implement `Default`. 

So I guess this struct is the best we can do with those constraints.